src/command.rs

```diff
@@ -1502,6 +1502,47 @@ impl ToJson for KeyDownAction {
 }
 
 #[derive(PartialEq)]
+pub enum PointerOrigin {
+    Viewport,
+    Pointer,
+    Element(WebElement),
+}
+
+impl Parameters for PointerOrigin {
+    fn from_json(body: &Json) -> WebDriverResult<PointerOrigin> {
+        match body {
+            &Json::String(ref x) => {
+                match &*x.to_string() {
```

Comment:
Is there any way to clean up `String(ref x)` and the `&*x.to_string()` that follows? I'll admit that I resorted to various combinations of ref, & and * until the compiler was happy. Will have to go and read how those actually work at some point.